### ISSUE TITLE
Color-coded categories

### ISSUE THREAD

**Issue by wada3n** [phase1geo/Outliner#48]
Color-coded categories allow you to add a second layer of structure to an outline, independent of item positions in the document.
In the app there is the possibility to highlight the text but it does not include the entire row.

e.g this image from outlineedit app on Mac:
![2020-05-31_10-06](https://user-images.githubusercontent.com/13304792/83349421-afb53b80-a32c-11ea-8ce9-e62d0251c3ee.png)

this will be important to create a visual impact of focus.

- As for the colored categories, they are organized, added and deleted through the status bar, as each outline has its own categories, and this categories is not related to all files.
- You can add default categories and choose the default category color scheme for new layouts from the app's preferences.

**Comment by phase1geo** [comment]
I feel like this functionality overlaps with the tags functionality. The main difference being that the entire row is highlighted and is only associated with one tag. I wonder if there is a way to combine the two approaches.

**Comment by wada3n** [comment]
Yes, I thought about this at first. But the idea relates to the visual aspect of organization,
While the tags is related to writing.
Unless the two ideas are combined and filtering is provided directly through it, specific colors are added to each tag, through the settings panel, either in the status bar or in the preferences. It would be a really beautiful thing.
Don't you think this will be too much for the tag?
Personalizing each idea will give an additional technical, and aesthetic dimension, I think so.
